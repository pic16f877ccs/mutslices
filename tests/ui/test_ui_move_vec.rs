use mutslices::MutSlice;

fn main() {
        let vec = Vec::<u8>::from([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
        let mut mut_slice = MutSlice::vec_into(vec);
        let ranges = [(0, 5), (5, 10), (10, 15)];
        let [ref mut one, ref mut two, ref mut three] = *mut_slice.slices_mut(&ranges) else { panic!(); };
        one.swap_with_slice(two);
        two.swap_with_slice(three);
        three.swap_with_slice(one);
        assert_eq!(vec, vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
}

