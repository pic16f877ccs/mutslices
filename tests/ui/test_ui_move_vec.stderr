error[E0382]: borrow of moved value: `vec`
  --> tests/ui/test_ui_move_vec.rs:11:9
   |
4  |         let vec = Vec::<u8>::from([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
   |             --- move occurs because `vec` has type `Vec<u8>`, which does not implement the `Copy` trait
5  |         let mut mut_slice = MutSlice::vec_into(vec);
   |                                                --- value moved here
...
11 |         assert_eq!(vec, vec![1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ value borrowed here after move
   |
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider cloning the value if the performance cost is acceptable
   |
5  |         let mut mut_slice = MutSlice::vec_into(vec.clone());
   |                                                   ++++++++
